addons/point_of_sale/static/src/js/Screens/TicketScreen/TicketScreen.js

```diff
@@ -16,6 +16,34 @@ odoo.define('point_of_sale.TicketScreen', function (require) {
             this.filter = null;
             this._initializeSearchFieldConstants();
         }
+        async willStart(){
+            await this._remove_completed_orders();
+        }
+        /**
+         * Remove unpaid orders that are synced to back-end from PoS UI
+         */
+        async _remove_completed_orders(){
+            var self = this;
+            var unpaid_orders = this.env.pos.db.get_unpaid_orders();
+            var orders = this.env.pos.get('orders').models;
+            unpaid_orders = unpaid_orders.map(function(e) {
+                return e.name
+            })
+            await this.env.pos.rpc({
+                model: 'pos.order',
+                method: 'check_synced_orders',
+                args: [unpaid_orders],
+            }, {
+                timeout: 30000 * unpaid_orders.length,
+            })
+            .then(function (order_ids) {
+                self.env.pos.db.remove_unpaid_order_ids(order_ids);
+                self.env.pos.get('orders').remove(orders.filter((order) => order_ids.includes(order.name)));
+            }).catch(function (reason){
+                var error = reason.message;
+                throw error;
+            });
```

Comment:
use try catch rather than `.then()` and `.catch()`
```suggestion
            try {
                const ordersRefsToRemove = await this.env.pos.rpc({
                    model: 'pos.order',
                    method: 'check_synced_orders',
                    args: [unpaid_orders],
                }, {
                    timeout: 30000,
                });
                this.env.pos.db.remove_unpaid_order_ids(ordersRefsToRemove);
                this.env.pos.get('orders').remove(orders.filter((order) => ordersRefsToRemove.includes(order.name)));
            } catch (e) {
                // here you show an ErrorPopup
            }
```